pupil-spa/src/app/feedback/feedback.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { StorageService } from '../services/storage/storage.service';
+
+@Component({
+  selector: 'app-feedback',
+  templateUrl: './feedback.component.html',
+  styleUrls: ['./feedback.component.scss']
+})
+export class FeedbackComponent implements OnInit {
+
+  public inputTypes: object;
+  public satisfactionRatings: object;
+  public selectedInputType;
+  public selectedSatisfactionRating;
+  public errorExists: boolean;
+  public errorInputType: boolean;
+  public errorSatisfactionRating: boolean;
+  public errorCommentExists: boolean;
+  public enableSubmit: boolean;
+
+  private pupilData: object;
+  private feedbackExists: boolean;
+  private feedbackData: object;
+  private submitted: boolean;
+
+  constructor(private router: Router, private storageService: StorageService) {}
+
+  ngOnInit() {
+    if (!this.validate()) {
+      this.router.navigate(['feedback-thanks']);
+    }
+    this.pupilData = this.storageService.getItem('pupil');
+    this.inputTypes = [
+      { id: 1, value: 'Touchscreen' },
+      { id: 2, value: 'Mouse' },
+      { id: 3, value: 'Keyboard' },
+      { id: 4, value: 'Mix of the above'}
+    ];
+    this.satisfactionRatings = [
+      { id: 1, value: 'Very easy' },
+      { id: 2, value: 'Easy' },
+      { id: 3, value: 'Neither easy or difficult' },
+      { id: 4, value: 'Difficult'},
+      { id: 5, value: 'Very difficult'}
+    ];
+    this.errorExists = false;
+    this.errorInputType = false;
+    this.errorSatisfactionRating = false;
+    this.errorCommentExists = false;
+    this.submitted = false;
+    this.enableSubmit = false;
+  }
+
+  validate() {
+    this.feedbackExists = this.storageService.getItem('feedback');
+    return (this.feedbackExists === null);
+  }
+
+  onSelectionChange(fieldType, fieldValue) {
+    switch (fieldType) {
+      case 'inputType':
+        this.selectedInputType = fieldValue;
+        break;
+      case 'satisfactionRating':
+        this.selectedSatisfactionRating = fieldValue;
+        break;
+    }
+
+    if (this.selectedInputType !== undefined && this.selectedSatisfactionRating !== undefined) {
+      this.enableSubmit = true;
+    }
+  }
+
+  onSubmit(commentText) {
+    if (this.submitted === true) {
+      return;
+    }
+    this.submitted = false;
```

Comment:
Shouldn't this be setting `this.submitted = true`?  Is there a unit test for duplicate submission?

Reply:
I don't see any test around this myself.
@pris54 do you think it should be covered?

Reply:
That's correct. ```this.submitted``` defaults to false and remains false until a successful submission happens (line 96).

Reply:
It was already set to false in `ngOnInit()` - why does it need to be set again?

Reply:
@jon-shipley True, line removed.